packages/hadron-build/lib/local-registry.js

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable no-console */
+const path = require('path');
+const execa = require('execa');
+const getPort = require('get-port');
+const waitOn = require('wait-on');
+const uuid = require('uuid');
+const os = require('os');
+const fs = require('fs');
+const util = require('util');
+
+const writeFile = util.promisify(fs.writeFile);
+const mkdir = util.promisify(fs.mkdir);
+const rmdir = util.promisify(fs.rmdir);
+
+const monorepoPath = path.dirname(require.resolve('../../../package.json'));
+
+function startServer(port, configPath) {
+  return execa('npx', [
+    'verdaccio',
+    '--config', path.basename(configPath), // circumvent any issues with windows paths
+    '--listen', port
+  ], { cwd: path.dirname(configPath), stdio: 'inherit' });
+}
+
+async function lernaPublish(registryAddress) {
+  await execa('npx', [
+    'lerna', 'publish', 'from-package',
+    '--ignore-scripts',
+    '--registry', registryAddress,
+    '--yes'
+  ], { cwd: monorepoPath, stdio: 'inherit' });
+}
+
+async function listPackages() {
+  const { stdout } = await execa('npx', [
+    'lerna', 'ls', '--json'
+  ], { cwd: monorepoPath });
+  return JSON.parse(stdout);
+}
+
+async function writeConfigFile(tempDir) {
+  // creates a verdaccio yaml config for the tmp folder
+  // NOTE: in order to disable the uplink with npm for internal packages
+  // we add overrides for each package.
```

Comment:
Not sure I understand why we need the overrides. `verdaccio` will not publish anything to remote and for installing it doesn't sound too bad to have them or am I missing something?

Reply:
yeah .. here the overrides serves a different purpose. With that setup we can publish the packages with same version (without bumping anything first), verdaccio won't even look for those on npm so `lerna publish from-package` won't detect them as already published.

Reply:
Ahh, it all makes sense now, thanks!